lib/commands/init.ts

```diff
@@ -0,0 +1,114 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import constants = require("./../constants");
+import helpers = require("./../common/helpers");
+import Future = require("fibers/future");
+import path = require("path");
+
+export class InitCommand implements ICommand {
+	private _projectFilePath: string;
+	
+	constructor(private $fs: IFileSystem,
+		private $errors: IErrors,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig,
+		private $projectHelper: IProjectHelper,
+		private $prompter: IPrompter,
+		private $npm: INodePackageManager,
+		private $npmInstallationManager: INpmInstallationManager) { }
+		
+	public allowedParameters: ICommandParameter[] = [];
+	public enableHooks = false;
+		
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let projectData: any = { };
+			
+			if(this.$fs.exists(this.projectFilePath).wait()) {
+				projectData = this.$fs.readJson(this.projectFilePath).wait();
+			}
+			
+			if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE]) {
+				projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE] = { };
+			}
+			
+			let projectDataBackup = _.extend({}, projectData);
+			
+			try {
+				this.$fs.writeJson(this.projectFilePath, projectData).wait(); // We need to create package.json file here in order to prevent "No project found at or above and neither was a --path specified." when resolving platformsData
```

Comment:
maybe this should be inside 
```TypeScript
if(!projectData[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE])
```
as this is the only case when we do not have package.json or it is missing required information. In all cases when we do not "enter" the mentioned if, we should not write the package.json again as it is already existing and correct.